include/aws/http/private/h1_encoder.h

```diff
@@ -37,10 +49,23 @@ enum aws_h1_encoder_state {
     AWS_H1_ENCODER_STATE_DONE,
 };
 
+enum aws_h1_encoder_body_stream_state {
+    AWS_H1_ENCODER_STATE_CHUNK_INIT,
+    AWS_H1_ENCODER_STATE_CHUNK_SIZE,
+    AWS_H1_ENCODER_STATE_CHUNK_EXTENSION_SEMICOLON_TOKEN,
+    AWS_H1_ENCODER_STATE_CHUNK_EXTENSION_KEY,
+    AWS_H1_ENCODER_STATE_CHUNK_EXTENSION_EQUAL_TOKEN,
+    AWS_H1_ENCODER_STATE_CHUNK_EXTENSION_VALUE,
```

Comment:
I was trying to copy as little memory as possible, but the size of  chunk extensions should normally be none or quite small. However, I do agree it will make the state machinery simpler and easier for the caller to. I will make this change.